### PR TITLE
[OF-1442] feat: Material Overrides

### PR DESCRIPTION
**Pull request by MAG-mv**
- Added material overrides props to static/animated model components
- Setup tests for static/animated model components
- put scripted properties on hold as this is not part of p0 and quickjs has issues with maps

### CODE REVIEW

File: Library/include/CSP/Multiplayer/Components/AnimatedModelSpaceComponent.h [hunk 1]
```diff
@@ -140,6 +141,22 @@ class CSP_API AnimatedModelSpaceComponent : public ComponentBase,
 	/// @return The index of the currently active animation.
 	void SetAnimationIndex(int64_t Value);
 
+	/// @brief Gets the material overrides of this component.
+	/// Should be in the format:
+	/// Key = Path to the model
+	/// Value = The material asset id
```
Comment (MAG-AdrianMeredith) [inline Library/include/CSP/Multiplayer/Components/AnimatedModelSpaceComponent.h:147]: I'm not sure we should refer to materials as "assets".  The fact that they are stored as assets isn't important (they could be stored anywhere).  I would personally prefer just saying **material id**.

File: Library/src/Multiplayer/Components/AnimatedModelSpaceComponent.cpp [hunk 2]
```diff
@@ -146,6 +148,41 @@ void AnimatedModelSpaceComponent::SetAnimationIndex(int64_t Value)
...
+	const auto& ReplicatedOverrides = GetMapProperty(static_cast<uint32_t>(AnimatedModelPropertyKeys::MaterialOverrides));
+	csp::common::Map<csp::common::String, csp::common::String> Overrides;
+
+	const auto* Keys = ReplicatedOverrides.Keys();
+
+	for (size_t i = 0; i < Keys->Size(); ++i)
+	{
+		const auto& CurrentKey			  = (*Keys)[i];
+		Overrides[CurrentKey.GetString()] = ReplicatedOverrides[CurrentKey].GetString();
+	}
```
[the remark this answers is not in the document]
Comment (MAG-mv) [reply Library/src/Multiplayer/Components/AnimatedModelSpaceComponent.cpp:163]: Agreed!

File: Tests/src/PublicAPITests/ComponentTests/AnimatedModelComponentTests.cpp [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+		const csp::common::Vector4 TestRotation(1.f, 1.f, 1.f, 1.f);
+		const csp::common::Vector3 TestScale(2.f, 2.f, 2.f);
+		const SpaceTransform TestTransform(TestPosition, TestRotation, TestScale);
+		constexpr const bool TestIsLoopPlayback			 = true;
+		constexpr const bool TestIsPlaying				 = true;
+		constexpr const int TestAnimationIndex			 = 1;
+		constexpr const bool TestIsVisible				 = false;
+		constexpr const bool TestIsARVisible			 = false;
+		constexpr const char* TestThirdPartyComponentRef = "TestThirdPartyComponentRef";
+		constexpr const bool TestIsShadowCaster			 = false;
```
[the remark this answers is not in the document]
Comment (MAG-mv) [reply Tests/src/PublicAPITests/ComponentTests/AnimatedModelComponentTests.cpp:103]: We can talk about this offline and try and resolve

File: Library/include/CSP/Multiplayer/Components/AnimatedModelSpaceComponent.h (hunk 1 above)
Comment (MAG-mv) [reply Library/include/CSP/Multiplayer/Components/AnimatedModelSpaceComponent.h:147]: I will also update the naming on the Material getters

File: Library/src/Multiplayer/Components/AnimatedModelSpaceComponent.cpp [hunk 4]
```diff
@@ -146,6 +148,41 @@ void AnimatedModelSpaceComponent::SetAnimationIndex(int64_t Value)
 	SetProperty(static_cast<uint32_t>(AnimatedModelPropertyKeys::AnimationIndex), Value);
 }
 
+csp::common::Map<csp::common::String, csp::common::String> AnimatedModelSpaceComponent::GetMaterialOverrides() const
+{
+	// Convert replicated values map to string values
+	const auto& ReplicatedOverrides = GetMapProperty(static_cast<uint32_t>(AnimatedModelPropertyKeys::MaterialOverrides));
```
[the remark this answers is not in the document]
Comment (MAG-mv) [reply Library/src/Multiplayer/Components/AnimatedModelSpaceComponent.cpp:154]: updated

File: Tests/src/PublicAPITests/ComponentTests/AnimatedModelComponentTests.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (MAG-ElliotMorris) [reply Tests/src/PublicAPITests/ComponentTests/AnimatedModelComponentTests.cpp:103]: Aye, a small thing I'd like to accomplish during our refactor period.

File: Tests/src/PublicAPITests/ComponentTests/AnimatedModelComponentTests.cpp [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+		EXPECT_EQ(AnimatedModelComponent->GetPosition(), TestPosition);
+		EXPECT_EQ(AnimatedModelComponent->GetRotation(), TestRotation);
+		EXPECT_EQ(AnimatedModelComponent->GetScale(), TestScale);
+		EXPECT_EQ(AnimatedModelComponent->GetIsLoopPlayback(), TestIsLoopPlayback);
+		EXPECT_EQ(AnimatedModelComponent->GetIsPlaying(), TestIsPlaying);
+		EXPECT_EQ(AnimatedModelComponent->GetAnimationIndex(), TestAnimationIndex);
+		EXPECT_EQ(AnimatedModelComponent->GetIsVisible(), TestIsVisible);
+		EXPECT_EQ(AnimatedModelComponent->GetIsARVisible(), TestIsARVisible);
+		EXPECT_EQ(AnimatedModelComponent->GetThirdPartyComponentRef(), TestThirdPartyComponentRef);
+		EXPECT_EQ(AnimatedModelComponent->GetIsShadowCaster(), TestIsShadowCaster);
```
[the remark this answers is not in the document]
Comment (MAG-ElliotMorris) [reply Tests/src/PublicAPITests/ComponentTests/AnimatedModelComponentTests.cpp:149]: Fair enough.

**Review by MAG-ElliotMorris — APPROVED** [review]
👍 Thanks for addressing feedback.

**Pull request merged**